src/kademlia/dht_storage.cpp

```diff
@@ -346,6 +345,7 @@ namespace
 		}
 
 		void put_immutable_item(sha1_hash const& target
+			, sha1_hash const& our_id
```

Comment:
I think this (option 3) could make difficult the life of other bindings (yes... java :D) but I will give it a try in code